plutus-core/cost-model/create-cost-model/CostModelCreation.hs

```diff
@@ -127,33 +127,42 @@ instance FromNamedRecord LinearModelRaw where
 
 instance FromRecord LinearModelRaw
 
+-- If we have data where the costs are nearly constant we can get a model with
+-- a negative slope, which can be problematic.  We use the `pos` function to
```

Comment:
This is kind of in flux at the moment, but I'll document it properly once it's settled down.

Reply:
I've now moved this into the R code: the models are examined and adjusted to replace any negative coefficients with 0, so Haskell should never see a model that can produce negative numbers.  It might be adding  check in the Haskell code just to be certain that that never occurs.  I'll explain the process in `CostModel.md`.